rslaser/rspulse/pulse.py

```diff
@@ -107,6 +134,16 @@ def __init__(self, params):
         self._sxvals = []  # horizontal slice data
         self._syvals = []  # vertical slice data
 
+    def get_params(self, params):
```

Comment:
get_params is mostly a copied function between LaserPulse, LaserPulseSlice, and LaserCavitiy. I think you can refactor to use a base class. If you'd like to do this setup time and  I can show you how I think it can be done.